### ISSUE TITLE
Can the ccache build instructions be simplified?

### ISSUE THREAD

**Issue by johnf** [facebook/react-native-website#4215]
## Description

https://reactnative.dev/docs/build-speed has detailed instructions for enabling ccace for xcode, I think this can be simplified

## What is the problem?

The current instructions indicate the following code should be added to the Podfile
```ruby

installer.pods_project.targets.each do |target|
      target.build_configurations.each do |config|
        # Using the un-qualified names means you can swap in different implementations, for example ccache
        config.build_settings["CC"] = "clang"
        config.build_settings["LD"] = "clang"
        config.build_settings["CXX"] = "clang++"
        config.build_settings["LDPLUSPLUS"] = "clang++"
      end
    end
```

## How can we address it?

I think this can be replace by uncommenting the `:ccache_enabled` argument to  `react_native_post_install`

e.g.
```ruby
    react_native_post_install(
      installer,
      config[:reactNativePath],
      :mac_catalyst_enabled => false,
      :ccache_enabled => true # this is normally commented out
    )
```

## Why is it important?

RN now supports caching and from a quick look at the code does a better job at it. A pod update even recommends enabling ccache_enabled. So the docs aren't in line with the tooling.

## Who needs this?

ios developers that want build peed improvements

## When should this happen (use version numbers if needed)?

I would suggest updating the docs for all versions of RN that support the ccache_enabled flag

## Other

I'd be happy to create a PR for the changes if the above is correct.

**Comment by cortinico** [comment]
I'll defer to @cipolleschi here

**Comment by cipolleschi** [comment]
Hi @johnf, could you open a PR with the change?

**Comment by johnf** [comment]
@cipolleschi Happy to!
Quick question about versioning. This change was implemented in 0.74. Does that mean I need to create versions in website/version_docs using the existing /docs for 0.72 and 0.73 and then update the various config files to support more versions?

**Comment by cipolleschi** [comment]
@johnf If the change has been introduced in 0.74, I think it's fine to just update `website/version_docs` for 0.74, 0.75 and the `docs` ones. We can leave the older versions as they are.
